src/Extensions/ElementalAreasExtension.php

```diff
@@ -278,17 +276,27 @@ public function supportsElemental()
      */
     public function ensureElementalAreasExist($elementalAreaRelations)
     {
+        $owner = $this->getOwner();
         foreach ($elementalAreaRelations as $eaRelationship) {
             $areaID = $eaRelationship . 'ID';
-
-            if (!$this->owner->$areaID) {
+            if (!$owner->$areaID) {
                 $area = ElementalArea::create();
                 $area->OwnerClassName = get_class($this->owner);
+                $writeAreaFunc = fn() => $area->write();
+                // Do not attempt to set the ElementalArea.TopPageID if the owner (e.g. Page)
+                // has not yet been persisted to the database, as this will cause a potentially
+                // large number of unneccessary database queries
+                if ($owner->ID === 0 && $area::has_extension(TopPageElementExtension::class)) {
```

Comment:
```suggestion
                if (!$owner->isInDB() && $area::has_extension(TopPageElementExtension::class)) {
```
This method exists specifically for these scenarios.

Reply:
Updated